### ISSUE TITLE
Make the bot protocol agnostic

### ISSUE THREAD

**Issue by meskarune** [HalosGhost/irc_bot#23]
Feature request to make the bot protocol agnostic. This way other protocols like telegram, slack, xmpp, etc could be added.

**Comment by HalosGhost** [comment]
Technically, it already is protocol agnostic. Though, all the plugins are geared towards the `irc` module (which is probably what you're referring to). Changing that is likely very difficult (especially since some functionality is likely to be IRC-specific.

Either way, #40 will likely be needed/wanted first.

**Comment by meskarune** [comment]
I think if plugins only return text strings, the bot itself can send the text using whatever protocol. Instead of having plugins do args.modules.irc.privmsg themselves the bot can do the thing. Maybe there should be some way to set what protocol a plugin can be used with. Something like convert can easily do xmpp and irc, but ircctl is irc only.

**Comment by HalosGhost** [comment]
Of course; but, for example, the moderate plugin is explicitly targeted at doing IRC moderation. That it might be able to talk on any protocol because it returns text wouldn't help the fact that the plugin is IRC-specific.
